lib/motors/controller.hpp

```diff
@@ -1,19 +1,35 @@
 #pragma once
+
 #include <cstdint>
+#include <optional>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include <rapidjson/document.h>
+#include <rapidjson/error/en.h>
+#include <rapidjson/istreamwrapper.h>
 
 #include "core/logger.hpp"
+#include "core/types.hpp"
 
 namespace hyped::motors {
 enum class ControllerStatus { kControllerTemperatureExceeded, kUnrecoverableWarning, kNominal };
 
 class Controller {
  public:
   Controller(core::ILogger &logger);
+  core::Result parseMessageFile(const std::string &path);
   void processErrorMessage(const std::uint16_t error_code);
   ControllerStatus processWarningMessage(const std::uint8_t warning_code);
```

Comment:
I'm a little bit unsure about what the purpose of this class and in particular the message file is, but based on my beliefs here's what I think the interface should be:

A constructor that takes the "list" (or other data structure) of configuration messages, a function (we've started calling them `create` so let's use that name) that parses the message file, collects the messages, constructs the object, and possibly sends the configuration messages. 

What I'm getting at: on one hand we don't want constructors to do I/O because it can fail and leave the object in an undefined state but on the other hand we don't want these magic function that the user has to remember to call (like `parseMessageFile`) in order to make the controller useful at all. There won't be more than one message file I believe, so only passing one should be fine.

Reply:
yea makes sense,

the message file exists so that we can change what messages get sent to the controller easily, and as EHW will lock our codebase at some point before the competition we can still update for example controller current limits or PID gains or similar